Android/Java/app/src/main/java/com/ironsource/ironsourcesdkdemo/DemoActivity.java

```diff
@@ -173,8 +191,8 @@ public void destroyBannerButtonTapped(View view) {
     private void destroyBanner() {
         if (bannerParentLayout != null) {
             log("destroyBanner");
-            IronSource.destroyBanner(ironSourceBannerLayout);
-            bannerParentLayout.removeView(ironSourceBannerLayout);
+            mBannerAd.destroy();
+            bannerParentLayout.removeView(mBannerAd);
             this.setBannerViewVisibility(View.GONE);
```

Comment:
I would move all the destroy logic to "onDestroy" function in the activity lifecycle, and in this way we don't need to handle the creation of a new object. (Max also doing that [https://github.com/AppLovin/AppLovin-MAX-SDK-Android/blob/master/AppLovin%20MAX%20Demo%20App%20-%20Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/ads/max/banner/ProgrammaticBannerAdActivity.java#L27](url) )

Reply:
not sure what you mean, the destory code you mentioned should not occur when the activity lifecycle onDestroy is called but when the user clicks to destroy the app.
The link you put is not working